src/vonage/video.py

```diff
@@ -163,33 +155,127 @@ def add_stream_to_archive(self, archive_id, stream_id, has_audio=True, has_video
             self._client.video_host(),
             f'/v2/project/{self._client.application_id}/archive/{archive_id}/streams',
             params,
-            auth_type=Video.auth_type
+            auth_type=Video.auth_type,
         )
-    
+
     def remove_stream_from_archive(self, archive_id, stream_id):
         params = {'removeStream': stream_id}
 
         return self._client.patch(
             self._client.video_host(),
             f'/v2/project/{self._client.application_id}/archive/{archive_id}/streams',
             params,
-            auth_type=Video.auth_type
+            auth_type=Video.auth_type,
         )
 
     def stop_archive(self, archive_id):
         return self._client.post(
             self._client.video_host(),
             f'/v2/project/{self._client.application_id}/archive/{archive_id}/stop',
             params=None,
-            auth_type=Video.auth_type
+            auth_type=Video.auth_type,
         )
 
     def change_archive_layout(self, archive_id, params=None, **kwargs):
         return self._client.put(
             self._client.video_host(),
             f'/v2/project/{self._client.application_id}/archive/{archive_id}/layout',
             params or kwargs,
-            auth_type=Video.auth_type
+            auth_type=Video.auth_type,
+        )
+
+    def create_sip_call(self, session_id: str, token: str, sip: dict):
+        if 'uri' not in sip:
+            raise SipError('You must specify a uri when creating a SIP call.')
+
+        params = {'sessionId': session_id, 'token': token, 'sip': sip}
+        return self._client.post(
+            self._client.video_host(),
+            f'/v2/project/{self._client.application_id}/dial',
+            params,
+            auth_type=Video.auth_type,
+        )
+
+    def play_dtmf(self, session_id: str, digits: str, connection_id: str = None):
+        if not re.search('^[0-9*#p]+$', digits):
+            raise InvalidInputError('Only digits 0-9, *, #, and "p" are allowed.')
+
+        params = {'digits': digits}
+
+        if connection_id is not None:
+            return self._client.post(
+                self._client.video_host(),
+                f'/v2/project/{self._client.application_id}/session/{session_id}/connection/{connection_id}/play-dtmf',
+                params,
+                auth_type=Video.auth_type,
+            )
+
+        return self._client.post(
+            self._client.video_host(),
+            f'/v2/project/{self._client.application_id}/session/{session_id}/play-dtmf',
+            params,
+            auth_type=Video.auth_type,
+        )
+
+    def list_broadcasts(self, offset=None, count=None, session_id=None):
+        params = {'offset': str(offset), 'count': str(count), 'sessionId': session_id}
```

Comment:
Validate that count is between 0 and 1000 (inclusive), and offset is >= 0.

Reply:
Have added these checks.